### PR TITLE
M1k53r/gamemaster

### PR DESCRIPTION
**Pull request by m1k53r**
> dude check this out new mikisheesh pr'ka just dropped
> sheeeeeesh

### CODE REVIEW

File: src/headers/GameMaster.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#include "headers/Scene.hpp"
+#include "headers/InputBox.hpp"
+
+class GameMaster : public Scene {
+private:
+    std::string name = "GameMasterScene";
+    smk::Window &window;
+    smk::Font font = smk::Font("/resources/SHPinscher-Regular.otf", 48);
+    std::map<std::string, bool> words;
```
Comment (gnwisniewski) [inline src/headers/GameMaster.hpp]: u mean passwords? 🙄🙄

Comment (m1k53r) [reply src/headers/GameMaster.hpp]: sorry mb :crying_cat_face:

File: src/scenes/GameMaster.cpp [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    window.Draw(codeText);
+
+    inputBox->drawText();
+
+    // For some reason nothing renders when inputBox
+    // is being drawn first
+    // Me from the future: 
+    // Ok so this happend because "draw" function 
+    // implemented some stuff for Game Creation scene
+    // which was not present in the current scene
```
Comment (gnwisniewski) [inline src/scenes/GameMaster.cpp:194]: my bad

File: src/scenes/GameMaster.cpp [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+#include <string>
+#include <locale>
+#include <codecvt>
+#include <algorithm>
+
+GameMaster::GameMaster(smk::Window &window) : window(window) {
+  this->inputBox = new InputBox(500, 150, 800, 50, window, 48);
+  inputBox->setColor(smk::Color::RGB(0.6, 0.4, 0.6));
+  page = 0;
+  words =
```
Comment (gnwisniewski) [inline src/scenes/GameMaster.cpp]: I don't get it. Do we have to manually add those words? What about working with presets?

Comment (m1k53r) [reply src/scenes/GameMaster.cpp]: It's there only for testing purposes. At the time I was making it, you haven't commited map with your presets yet so I made my own. When we merge it, I will remove it.

**Pull request merged**